src/Talis/Persona/Client/Tokens.php

```diff
@@ -7,13 +7,42 @@ class ScopesNotDefinedException extends \Exception
 {
 }
 
+function setCookieFuncDefault($token) {
+    if (headers_sent()) {
+        $this->getLogger()->error(
+          'Cannot set access token within cookie as ' .
+          'the headers have already been sent'
+        );
+    } else {
+        $this->getLogger()->debug('Setting access token in cookie');
+
+        setcookie(
+            'access_token',
+            json_encode($token),
+            time() + $token['expires_in']
+        );
+    }
+};
+
 class Tokens extends Base
 {
     /**
      * Cached connection to redis
      * @var \Predis\Client
      */
     protected $tokenCacheClient = null;
+    private $setCookieFunc;
+
+    public function __construct(array $config, $setCookieFunc = null)
```

Comment:
It's not a breaking change when `$setCookieFunc` is optional. This is a backwards-compatible change so stick to a 3.x.x version.